clippy_lints/src/array_indexing.rs

```diff
@@ -117,13 +116,13 @@ fn to_const_range(
     let start = match *start {
         Some(Some(ConstVal::Integral(x))) => x,
         Some(_) => return None,
-        None => ConstInt::Infer(0),
+        None => ConstInt::U8(0),
```

Comment:
This logic is not possible anymore. We need to use `to_u64_unchecked` and `is_signed` to reimplement it with integers instead of `ConstInt`